src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt

```diff
@@ -277,4 +275,58 @@ class SessionService(
       sessionDatesUtil.isBiWeeklySessionActiveForDate(date, sessionTemplate)
     }
   }
+
+  fun getSessionSchedule(prisonCode: String, sessionDate: LocalDate): List<SessionScheduleDto> {
+
+    var sessionTemplates = sessionTemplateRepository.findValidSessionTemplatesForSession(
+      prisonCode,
+      sessionDate,
+      sessionDate.dayOfWeek
+    )
+
+    sessionTemplates = filterSessionsTemplatesForDate(sessionDate, sessionTemplates)
+
+    return sessionTemplates.map { sessionTemplate -> createSessionInfoDto(sessionTemplate) }.toList()
+  }
+
+  private fun createSessionInfoDto(sessionTemplate: SessionTemplate): SessionScheduleDto {
+    val sessionTemplateFrequency = getSessionTemplateFrequency(sessionTemplate)
+
+    return SessionScheduleDto(
+      sessionTemplateReference = sessionTemplate.reference,
+      startTime = sessionTemplate.startTime,
+      endTime = sessionTemplate.endTime,
+      capacity = SessionCapacityDto(sessionTemplate),
+      prisonerLocationGroupNames = sessionTemplate.permittedSessionGroups.map { it.name }.toList(),
+      sessionTemplateFrequency = sessionTemplateFrequency,
+      sessionTemplateEndDate = sessionTemplate.validToDate
+    )
+  }
+
+  private fun getSessionTemplateFrequency(sessionTemplate: SessionTemplate): SessionTemplateFrequency {
+
+    val firstSessionDate = adjustDateByDayOfWeek(sessionTemplate.dayOfWeek, sessionTemplate.validFromDate)
+
+    if (isNotMoreThanAWeek(firstSessionDate, sessionTemplate)) {
+      return SessionTemplateFrequency.ONE_OFF
+    }
+    if (sessionTemplate.biWeekly) {
+      return SessionTemplateFrequency.BI_WEEKLY
+    }
+    return SessionTemplateFrequency.WEEKLY
+  }
+
+  private fun isNotMoreThanAWeek(
+    firstSessionDate: LocalDate,
+    sessionTemplate: SessionTemplate
+  ): Boolean {
+    return sessionTemplate.validToDate != null && DAYS.between(firstSessionDate, sessionTemplate.validToDate) < 8
```

Comment:
should this be < 7 for it to be one off?

Reply:
I think it has to be < 8 e.g.
MON (1), TUS (2), WEN (3), THU (4), FRI(5), SAT(6), SUN(7), MON(8) therefore less than 8 I think right?

Reply:
I'm being silly it's MON (0), TUS (1), WEN (2), THU (3), FRI(4), SAT(5), SUN(6), MON(7)